### ISSUE TITLE
Throws Uncaught ReferenceError: exports is not defined

### ISSUE THREAD

**Issue by chrisalexander55** [weichx/cerialize#54]
Excellent library and thank you for sharing with the world!

So, let me give you my config:

OS: OSX 10.12.3
Node: v6.10.0
Angular-CLI: 1.0.0

package.json dependencies node:
`"dependencies": {
    "@angular/common": "^4.0.0",
    "@angular/compiler": "^4.0.0",
    "@angular/core": "^4.0.0",
    "@angular/forms": "^4.0.0",
    "@angular/http": "^4.0.0",
    "@angular/platform-browser": "^4.0.0",
    "@angular/platform-browser-dynamic": "^4.0.0",
    "@angular/router": "^4.0.0",
    "bootstrap": "^3.3.7",
    "cerialize": "^0.1.14",
    "core-js": "^2.4.1",
    "hammerjs": "^2.0.8",
    "jquery": "^3.2.1",
    "moment": "^2.18.1",
    "rxjs": "^5.1.0",
    "zone.js": "^0.8.4"
  },`

.angular-cli.json scripts node:
`"scripts": [
        "../node_modules/jquery/dist/jquery.min.js",
        "../node_modules/bootstrap/dist/js/bootstrap.min.js",
        "../node_modules/moment/min/moment.min.js",
        "../node_modules/cerialize/dist/serialize.js"
      ],`

tsconfig.json:
`{
  "compileOnSave": false,
  "compilerOptions": {
    "outDir": "./dist/out-tsc",
    "baseUrl": "src",
    "sourceMap": true,
    "declaration": false,
    "moduleResolution": "node",
    "emitDecoratorMetadata": true,
    "experimentalDecorators": true,
    "target": "es5",
    "typeRoots": [
      "node_modules/@types"
    ],
    "lib": [
      "es2016",
      "dom"
    ]
  }
}`

tsconfig.app.json:
`{
  "compileOnSave": false,
  "compilerOptions": {
    "outDir": "./dist/out-tsc",
    "baseUrl": "src",
    "sourceMap": true,
    "declaration": false,
    "moduleResolution": "node",
    "emitDecoratorMetadata": true,
    "experimentalDecorators": true,
    "target": "es5",
    "typeRoots": [
      "node_modules/@types"
    ],
    "lib": [
      "es2016",
      "dom"
    ]
  }
}`

Here is the error:
`Uncaught ReferenceError: exports is not defined
    at eval (eval at webpackJsonp.91.module.exports (addScript.js:9), <anonymous>:11:1)
    at eval (<anonymous>)
    at webpackJsonp.91.module.exports (addScript.js:9)
    at Object.418 (serialize.js?eb3b:1)
    at __webpack_require__ (bootstrap 6f84d9f…:52)
    at Object.889 (scripts.bundle.js:65)
    at __webpack_require__ (bootstrap 6f84d9f…:52)
    at webpackJsonpCallback (bootstrap 6f84d9f…:23)
    at scripts.bundle.js:1`

A little heavy on the detail but I wanted to provide you with my environment as best as possible. When I remove cerialize from my .angular-cli.json scripts node, the error vanishes. Did I uncover a bug or did I misconfig something? Thanks!

**Comment by weichx** [comment]
I don't think this has anything to do with config, I haven't used angular2/ angluar-cli / whatever but I know they use system.js instead of commonjs. (at least they used to, im behind the time on the angular side of things). It could be that they are loading using the system packer and Cerialize expects commonjs style imports. You could verify this by importing the raw ts file (serialize.ts) from the Cerialize library instead of the pre-compiled version. Let me know if that helps

**Comment by chrisalexander55** [comment]
@weichx I perform 2 follow-ups:

1. I changed compilerOptions to module: commonjs and it worked without incident. Thank you.
2. I imported the serizlize.ts file directly and it threw the following

`ERROR in /Users/jamesalexander/Documents/Projects/cms-ide/node_modules/cerialize/src/serialize.ts (501,45): Argument of type 'string | Date' is not assignable to parameter of type 'string'.
  Type 'Date' is not assignable to type 'string'.
/Users/jamesalexander/Documents/Projects/cms-ide/node_modules/cerialize/src/serialize.ts (544,25): Argument of type 'string | Date' is not assignable to parameter of type 'string'.
  Type 'Date' is not assignable to type 'string'.
/Users/jamesalexander/Documents/Projects/cms-ide/node_modules/cerialize/src/serialize.ts (642,42): Argument of type 'string | Date' is not assignable to parameter of type 'string'.
  Type 'Date' is not assignable to type 'string'.`

Although the commonjs module format will allow me to move forward, just letting you know of what I found. Truthfully, I'd rather use es2015 modules as that is where JS is moving, IMHO.
